client/app/bundles/course/experience-points/disbursement/operations.ts

```diff
@@ -0,0 +1,253 @@
+        data[field],
+      );
+    }
+  });
+
+  data.pointList.forEach((point: PointListData, index) => {
+    if (point.id) {
+      payload.append(
+        `experience_points_forum_disbursement[experience_points_records_attributes][${index}][points_awarded]`,
+        point.points,
+      );
+      payload.append(
+        `experience_points_forum_disbursement[experience_points_records_attributes][${index}][course_user_id]`,
+        point.id?.toString() ?? '',
+      );
+    }
+  });
+  return payload;
+};
+
+const formatFilterAttribute = (
+  filter: ForumDisbursementFilters,
+): ForumDisbursementFilterParams => ({
+  params: {
+    'experience_points_forum_disbursement[start_time]': filter.startTime,
+    'experience_points_forum_disbursement[end_time]': filter.endTime,
+    'experience_points_forum_disbursement[weekly_cap]': filter.weeklyCap,
+  },
+});
+
+const formatSearchAttribute = (
+  filter: ForumDisbursementFilters,
+  user: ForumDisbursementUserData,
+): ForumSearchParams => ({
+  params: {
+    'search[course_user_id]': user.id,
+    'search[start_time]': filter.startTime,
+    'search[end_time]': filter.endTime,
+  },
+});
+
+export function fetchDisbursements(): Operation<
+  AxiosResponse<{
+    currentGroup: CourseGroupListData | null;
+    courseGroups: CourseGroupListData[];
+    courseUsers: CourseUserListData[];
+  }>
+> {
+  return async (dispatch) =>
+    CourseAPI.disbursement
+      .index()
+      .then((response) => {
+        const data = response.data;
+        dispatch(
+          actions.saveDisbursementList(
+            data.currentGroup,
+            data.courseGroups,
+            data.courseUsers,
+          ),
+        );
+        return response;
+      })
+      .catch((error) => {
+        throw error;
+      });
+}
+
+export function fetchForumDisbursements(): Operation<
+  AxiosResponse<{
+    filters: ForumDisbursementFilters;
+    forumUsers: ForumDisbursementUserData[];
+  }>
+> {
+  return async (dispatch) =>
+    CourseAPI.disbursement
+      .forumIndex()
+      .then((response) => {
+        const data = response.data;
+        dispatch(
+          actions.saveForumDisbursementList(data.filters, data.forumUsers),
+        );
+        return response;
+      })
+      .catch((error) => {
+        throw error;
+      });
+}
+
+export function fetchFilteredForumDisbursements(
+  filter: ForumDisbursementFilters,
+): Operation<
+  AxiosResponse<{
+    filters: ForumDisbursementFilters;
+    forumUsers: ForumDisbursementUserData[];
+  }>
+> {
+  const filterAttributes: ForumDisbursementFilterParams =
+    formatFilterAttribute(filter);
+
+  return async (dispatch) =>
+    CourseAPI.disbursement
+      .fetchForumIndexWithFilter(filterAttributes)
+      .then((response) => {
+        const data = response.data;
+        dispatch(
+          actions.saveForumDisbursementList(data.filters, data.forumUsers),
+        );
+        const users = data.forumUsers;
```

Comment:
Good try here!
A drawback: if there are many students, we are making too many concurrent requests. Should have just done a single api call.

But anyway, we dont want to fetch the forum posts until a user clicks on the modal. So you need to refactor this behaviour.